examples/ddpg_pmsm_dq_current_control.py

```diff
@@ -20,83 +20,138 @@
 from gym.core import Wrapper
 from gym.spaces import Box, Tuple
 
+# This example shows how we can use GEM to train a reinforcement learning agent to control the current within
+# a permanent magnet synchronous motor three-phase drive
+# We use a deep-deterministic-policy-gradient (DDPG) agent
+# to determine which action must be taken on a continuous-control-set
+
 
 class AppendLastActionWrapper(Wrapper):
     """
     The following environment considers the dead time in the real-world motor control systems.
-    The real-world system changes its state, while the agent calculates the next action based on a previoulsly measured
-    observation. Therefore, for the agents it seems as if the applied action effects the state one step delayed.
-    (with a dead time of one time-step)
+    The real-world system changes its state, while the agent simultaneously calculates the next action based on a
+    previously measured observation.
+    Therefore, for the agents it seems as if the applied action affects the environment with one step delay
+    (with a dead time of one time step).
 
-    For complete observability of the system at each time-step we append the last played action of the agent to the
-    observation, because this action will be the one that is active in the next step.
+    For complete observability of the system at each time step we append the last selected action to the
+    observation, because this action will be the one that is active while the agent has to make the next decision.
     """
     def __init__(self, environment):
         super().__init__(environment)
+        # append the action space dimensions to the observation space dimensions
         self.observation_space = Tuple((Box(
             np.concatenate((environment.observation_space[0].low, environment.action_space.low)),
             np.concatenate((environment.observation_space[0].high, environment.action_space.high))
         ), environment.observation_space[1]))
 
     def step(self, action):
+
         (state, ref), rew, term, info = self.env.step(action)
+
+        # extend the output state by the selected action
         state = np.concatenate((state, action))
+
         return (state, ref), rew, term, info
 
     def reset(self, **kwargs):
+
         state, ref = self.env.reset()
+
+        # extend the output state by zeros after reset
+        # no action can be appended yet, but the dimension must fit
         state = np.concatenate((state, np.zeros(self.env.action_space.shape)))
+
         return state, ref
 
 
 if __name__ == '__main__':
+
     tf.compat.v1.disable_eager_execution()
 
     window_length = 1

```

Comment:
@atra94 
How can we easily explain the "window_length" that kerasRL2 uses?

Reply:
Those are basically just the last "window_length" observations that are used as input to the network. Therefore, not only o_k is used as input but also o_k-1, o_k-2,... . The actor and critic observation inputs are therefore shaped (window_length, observation_length)


Initially, we tried to use a longer window length instead of the AppendLastActionWrapper. However, the latter one worked better to compensate for the dead time in the system. 
Maybe we can leave this line out and hard code the value everywhere it is used.

Reply:
I tried to include a short explanation of the window_length. It was used too often to hard code it, for my perception.
(3 times: actor, critic and experience replay buffer)

Reply:
This is the appearance in the code now:
```
# Define how many past observations we want the control agent to process each step
# for this case, we assume to pass only the single most recent observation
window_length = 1
```